### DISCUSSION TITLE
showOnPreview ignored where used in new function fieldsForPreview

### DISCUSSION THREAD

**Discussion by emkcloud** [laravel/nova-issues#4925] [Q&A] [answered]
Laravel Version: 9.33.0
Nova Version: 4.16.1
PHP Version: 8.1.7
Database Driver & Version: MYSQL 5.7
Operating System and Version: Ubuntu 22.04.1 LTS
Browser type and version: Brave Versión 1.37.116 Chromium: 100.0.4896.127 (64 bits)

### Description:
ShowOnPreview ignored where used in new function fieldsForPreview

```
public function fieldsForPreview(NovaRequest $request)
{
     return
     [
            Heading::make(Str::upper(__('labels.gift')))
               ->showOnPreview(function(){ return ($this->gifts->count() > 0 ); }),
     ]
}
```
In previous version i not see emprty section, now yes

![image](https://user-images.githubusercontent.com/5731295/195945108-7ac9c954-bd23-4c6d-ae52-2250c78db222.png)

**Comment by crynobone** [comment, marked as answer]
You should be able to do the following:

```php
public function fieldsForPreview(NovaRequest $request)
{
    return [
        $this->mergeWhen($this->gifts->count() > 0, function () {
            return [
                Heading::make(Str::upper(__('labels.gift'))),
            ];
        }),
    ];
}
```

**Comment by emkcloud** [comment]
Ok perfect

![image](https://user-images.githubusercontent.com/5731295/195966326-c466d97b-c593-4a94-b189-103fc6e60584.png)
